pkg/server/endpoints/endpoints_test.go

```diff
@@ -285,31 +303,55 @@ func TestListenAndServe(t *testing.T) {
 	})
 
 	t.Run("Agent", func(t *testing.T) {
-		testAgentAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testAgentAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 	t.Run("Debug", func(t *testing.T) {
-		testDebugAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testDebugAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 	t.Run("Health", func(t *testing.T) {
-		testHealthAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testHealthAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 	t.Run("Bundle", func(t *testing.T) {
-		testBundleAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testBundleAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 	t.Run("Entry", func(t *testing.T) {
-		testEntryAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testEntryAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 	t.Run("SVID", func(t *testing.T) {
-		testSVIDAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testSVIDAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 	t.Run("TrustDomain", func(t *testing.T) {
-		testTrustDomainAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+		testTrustDomainAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, federatedAdminConn, downstreamConn)
 	})
 
 	t.Run("Access denied to remote caller", func(t *testing.T) {
 		testRemoteCaller(ctx, t, target)
 	})
 
+	t.Run("Invalidate connection with misconfigured foreign admin caller", func(t *testing.T) {
+		unauthenticatedConfig := tlsconfig.MTLSClientConfig(unauthenticatedForeignAdminSVID, ca.X509Bundle(), tlsconfig.AuthorizeID(serverID))
+		unauthorizedConfig := tlsconfig.MTLSClientConfig(unauthorizedForeignAdminSVID, ca.X509Bundle(), tlsconfig.AuthorizeID(serverID))
+		unfederatedConfig := tlsconfig.MTLSClientConfig(unfederatedForeignAdminSVID, ca.X509Bundle(), tlsconfig.AuthorizeID(serverID))
+
+		for _, config := range []*tls.Config{unauthenticatedConfig, unauthorizedConfig, unfederatedConfig} {
+			conn, err := grpc.DialContext(ctx, endpoints.TCPAddr.String(),
+				grpc.WithTransportCredentials(credentials.NewTLS(config)),
+			)
+			require.NoError(t, err)
+
+			_, err = entryv1.NewEntryClient(conn).ListEntries(ctx, nil)
+			require.Error(t, err)
```

Comment:
Instead of actually issuing an RPC we can use grpc.FailOnNonTempDialError(true) and grpc.WithReturnConnectionError() to cause the dial to fail when the handshake is unsuccessful.